src/query/settings/src/settings_default.rs

```diff
@@ -775,6 +775,12 @@ impl DefaultSettings {
                     desc: "The maximum count of set operator in a query. If your query stack overflow, you can reduce this value.",
                     mode: SettingMode::Both,
                     range: Some(SettingRange::Numeric(0..=u64::MAX)),
+                }),
+                ("enable_loser_tree_merge_sort", DefaultSettingValue {
+                    value: UserSettingValue::UInt64(1),
```

Comment:
I still did not find `loser_tree` is faster than `heap`.

```
🐳 :) select l_orderkey from lineitem order by l_orderkey ignore_result;
0 row read in 1.409 sec. Processed 59.99 million row, 457.66 MiB (42.58 million rows/s, 324.82 MiB/s)

🐳 :) set enable_loser_tree_merge_sort = 0;
🐳 :) select l_orderkey from lineitem order by l_orderkey ignore_result;
0 row read in 1.258 sec. Processed 59.99 million row, 457.66 MiB (47.7 million rows/s, 363.90 MiB/s)
```

```

🐳 :) set enable_loser_tree_merge_sort = 1;
🐳 :) select userid from hits order by userid ignore_result;
0 row read in 3.027 sec. Processed 100 million row, 762.92 MiB (33.04 million rows/s, 252.08 MiB/s)
🐳 :) set enable_loser_tree_merge_sort = 0;
🐳 :) select userid from hits order by userid ignore_result;
0 row read in 2.528 sec. Processed 100 million row, 762.92 MiB (39.56 million rows/s, 301.83 MiB/s)
```

Reply:
Maybe we should disable it by default unless we find it's faster.

But the sort algorithm abstraction is fine.

Reply:
Is it true that when K is large enough, the performance of the loser tree will be better?

Current K is the num of physical cores by default.

Reply:
https://github.com/datafuselabs/databend/blob/7b15120fd2b2f91cf3a3f9ca860ea733f830eefc/src/query/pipeline/transforms/src/processors/transforms/sort/merger.rs#L171
I think the problem is with this peek_top2.
It is generally accepted that the loser tree is twice as fast as the heap, but the need to fetch top2 here causes the loser tree to be accessed twice per iteration. And the standard library's heap is deeply optimized. Maybe some kind of variant implementation can be found later to improve this situation.

Reply:
Tracked in https://github.com/datafuselabs/databend/issues/15925